cmd/update.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright © 2020 Luke Hinds <lhinds@redhat.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+
+	tcrypto "github.com/google/trillian/crypto"
+	"go.uber.org/zap"
+
+	tclient "github.com/google/trillian/client"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/merkle"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/projectrekor/rekor-cli/log"
+	"github.com/spf13/viper"
+
+	"github.com/spf13/cobra"
+)
+
+type latestResponse struct {
+	Proof *trillian.GetLatestSignedLogRootResponse
+	Key   []byte
+}
+
+type state struct {
+	Size int64
+	Hash []byte
+}
+
+func stateDir() (string, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	rekorDir := filepath.Join(home, ".rekor")
+	if _, err := os.Stat(rekorDir); os.IsNotExist(err) {
+		if err := os.Mkdir(rekorDir, 0755); err != nil {
+			return "", err
+		}
+	}
+	return filepath.Join(home, ".rekor", "rekor.json"), nil
+}
+
+func getPreviousState(log *zap.SugaredLogger) *state {
+	p, err := stateDir()
+	if err != nil {
+		return nil
+	}
+	var oldState state
+	f, err := ioutil.ReadFile(p)
+	if err != nil {
+		log.Info(err)
+		return nil
+	}
```

Comment:
This is really good Dan, making some great progress here.

A tiny nit: First time running we get a `.rekor/rekor.json: no such file or directory`. Its are capturing the err as it should and is correct and proper, but I guess we could put a pretty message there instread, as a file not existing is expected. I will merge and follow up.